### ISSUE TITLE
Overflow flag is mishandled

### ISSUE THREAD

**Issue by joshudson** [YJDoc2/8086-Emulator#14]
I tried:

start:
    MOV AX, 0xFFFF
    ADD AX, AX

Expected: AX = 0xFFFE, CF = 1, OF = 0
Got: AX = 0xFFFE, CF = 1, OF = 1

My understanding from Peter Cordes's analysis is OF is always tracking CF which is not right. https://stackoverflow.com/a/75679804/14768

**Comment by YJDoc2** [comment]
ref : https://github.com/YJDoc2/8086-emulator-web/issues/30

**Comment by YJDoc2** [comment]
Hey @joshudson , indeed there is a bug in the overflow flag handling :sweat_smile: 
Currently it always copies the carry flag, however the actual logic should be :

Set overflow flag when :
- 7/15 th bit generates carry, but 8/16 th doesn't
- 7/15 th bit does not generate carry, but 8/16 th does

Else clear.

There is a similar bug in setting the zero flag as well.

This was reported a few days back separately to me, and I have implemented the fix for it, however I got a bit busy and haven't done final validation+deploy for that fix.

I hope to get that done by this weekend.

Thanks a lot for using and reporting this, hope you found this useful (apart from the bug!)
